channeldb/control_tower.go

```diff
@@ -201,29 +285,34 @@ func (p *paymentControl) Fail(paymentHash [32]byte) error {
 		// meaning it never should have left the switch.
 		case paymentStatus == StatusGrounded:
 			updateErr = ErrPaymentNotInitiated
+			return nil
 
 		// A failed response was received for an InFlight payment, mark
 		// it as Failed to allow subsequent attempts.
 		case paymentStatus == StatusInFlight:
-			return bucket.Put(paymentStatusKey, StatusGrounded.Bytes())
 
 		// The payment was completed previously, and we are now
 		// reporting that it has failed. Leave the status as completed,
 		// but alert the user that something is wrong.
 		case paymentStatus == StatusCompleted:
 			updateErr = ErrPaymentAlreadyCompleted
+			return nil
 
 		// The payment was already failed, and we are now reporting that
 		// it has failed again. Leave the status as failed, but alert
 		// the user that something is wrong.
 		case paymentStatus == StatusFailed:
 			updateErr = ErrPaymentAlreadyFailed
+			return nil
 
 		default:
 			updateErr = ErrUnknownPaymentStatus
+			return nil
 		}
 
-		return nil
+		// A failed response was received for an InFlight payment, mark
```

Comment:
Would be easier to read IMO if this was lifted back up to the current passthrough case.

Reply:
No longer applicable.